extensions/support/xalan/runtime/src/main/java/org/apache/camel/quarkus/support/xalan/XalanSupport.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.support.xalan;
+
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class XalanSupport {
+    private static final Logger LOGGER = LoggerFactory.getLogger(XalanSupport.class);
+
+    private XalanSupport() {
+    }
+
+    public static TransformerFactory newTransformerFactoryInstance() {
+        TransformerFactory factory = TransformerFactory.newInstance(
+                "org.apache.xalan.xsltc.trax.TransformerFactoryImpl",
+                Thread.currentThread().getContextClassLoader());
```

Comment:
This looks like the key part of the solution. If so, it would be nice to add a short comment explaining what problem is this snippet solving so that the future readers do not need to wonder.

Reply:
Should I expand the description of the problem [here](https://github.com/apache/camel-quarkus/issues/427) ?

Reply:
Question: we cannot pass `org.apache.xalan.xsltc.trax.TransformerFactoryImpl` as a standard service provider, right? We'd thus get the right implementation, but the class loader would still be wrong, right?

Reply:
Probably we can use a standard service provider but that's the most "secure" way as you can't override it with a system property. If you know a better way please go ahead and send a PR

Reply:
> Should I expand the description of the problem [here](https://github.com/apache/camel-quarkus/issues/427) ?

I meant a comment in the above newTransformerFactoryInstance() method. It might not be clear to the future readers why it is there. A single sentence would be enough, something like `This solves the translet class loading problems on Java 11+, see https://github.com/apache/camel-quarkus/issues/427`